src/main/scala/mimir/algebra/Primitives.scala

```diff
@@ -239,3 +243,20 @@ case class NullPrimitive()
   def asDateTime: DateTime = throw new NullTypeException(TAny(), TDate(), "Hard Cast")
   def payload: Object = null
 }
+/**
+ * Boxed representation of Interval
+ */
+@SerialVersionUID(100L)
+case class IntervalPrimitive(ms: Long)
+  extends PrimitiveValue(TInterval())
+{
```

Comment:
There are several different notions of Intervals.  Irritatingly, terminology used for them differs based on who you're talking to.  The key distinction is that units of time above 1s are not interchangeable.  For example: 
* 1 hour = 3600s (Except during leap hours)
* 1 month = 28, 29, 30, or 31 days.

If I say `DATE('2016-02-03') + INTERVAL('1 month')` in a SQL query, I expect to get `{d'2016-03-03'}` and not `{d'2016-03-04'}` (assumption of 1 month = 30 days), `{d'2016-03-02'}` (forgetting leap months), etc...  

SQL makes the unit distinction clear.  Intervals are stored with units separated: This is analogous to what JODA calls a [Period](http://www.joda.org/joda-time/key_period.html).  There are enough corner cases that you shouldn't be trying to figure this out yourself.  Use a JODA Period instead of a Long.